kernel/generic/src/ipc/irq.c

```diff
@@ -322,7 +322,7 @@ static kobject_ops_t irq_kobject_ops = {
  *
  */
 errno_t ipc_irq_subscribe(answerbox_t *box, inr_t inr, sysarg_t imethod,
-    irq_code_t *ucode, cap_irq_handle_t *uspace_handle)
+    uspace_ptr(irq_code_t) ucode, uspace_ptr(cap_irq_handle_t) uspace_handle)
```

Comment:
One of the issues I see is that we are losing a degree of type safety here. Can we do something about that by expanding uspace_ptr into some opaque pointers?